tools/src/main/scala/scala/scalanative/build/LLVM.scala

```diff
@@ -31,44 +24,36 @@ private[scalanative] object LLVM {
    *  @return
    *    The paths of the `.o` files.
    */
-  def compile(config: Config, paths: Seq[Path]): Seq[Path] = {
+  def compile(
+      config: Config,
+      paths: Seq[Path]
+  ): Seq[CompilationResult] = {
+    import NativeSourcePlugin._
+
+    val plugins = LlSourcePlugin +: config.compilerConfig.nativeSourcePlugins
```

Comment:
According to the previous comment - if we would only define additional plugins in NativeConfig we should here define also CLang and Cpp plugins. Maybe, if in the future someone would like to introduce better Cpp integration we could add settings to nativeConfig allowing for disabling our internal implementations.

Reply:
See https://github.com/scala-native/scala-native/pull/2572#discussion_r812598777.